cores/esp8266/coredecls.h

```diff
@@ -17,14 +17,18 @@ extern bool timeshift64_is_set;
 void esp_yield();
 void esp_schedule();
 void tune_timeshift64 (uint64_t now_us);
-void settimeofday_cb (void (*cb)(void));
 void disable_extra4k_at_link_time (void) __attribute__((noinline));
 
 uint32_t sqrt32 (uint32_t n);
 uint32_t crc32 (const void* data, size_t length, uint32_t crc = 0xffffffff);
 
 #ifdef __cplusplus
 }
+
+#include <functional>
+void settimeofday_cb (std::function<void()>&& cb);
```

Comment:
Considering the number of repetitions I see, I think the arg type here should get a type alias, e.g.:
```cpp
using timeOfDayCB = std::function<void()>;
void setTimeOfDayCB (timeOfDayCB &&cb);
```

Reply:
Also, please camelCase.

Reply:
`settimeofday()` is a libc function.
Using `setTimeOfDayCB()` makes it quite unrelated with the system function.